test/new-e2e/tests/agent-platform/common/agent_integration.go

```diff
@@ -66,3 +68,21 @@ func CheckIntegrationInstall(t *testing.T, client *TestClient) {
 		require.Error(tt, err, "should raise error when trying to install version older than the one shipped")
 	})
 }
+
+func installIntegration(t *testing.T, client *TestClient, integration string) {
+	// This operation can fail if the release pipeline in integrations-core is running.
+	// The wheels are not generated in an atomic way and installation can fail during this process.
+	// This is a first step, we could also install the integration from a local wheel to avoid the problem altogether but that's not what customers do.
+	// Wheels can be unavailable for a few minutes.
+	// Their release pipeline runs at least once a day at 5AM CET and can run during working hours if they need to release something.
+
+	interval := 30 * time.Second
+	maxRetries := 6
+
+	err := backoff.Retry(func() error {
+		_, err := client.AgentClient.IntegrationWithError(agentclient.WithArgs([]string{"install", "-r", integration}))
+		return err
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(interval), uint64(maxRetries)))
+
+	require.NoError(t, err)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Instead of backoff you can use testify `Eventually`

Reply:
Having a look!